test/unit-tests/decorators.ts

```diff
@@ -170,5 +166,249 @@ describe("decorators", () => {
 				assert.deepEqual(err.message, expectedResults[index].message);
 			}));
 		});
+
+		describe("postAction", () => {
+			let isPostActionExecuted = false;
+			let isActionExecuted = false;
+			let expectedResults: any;
+
+			let postAction = () => {
+				if (!isActionExecuted) {
+					throw new Error("Post Action MUST be executed after all actions are executed.");
+				}
+				isPostActionExecuted = true;
+			};
+
+			let getPromisesWithPostAction = (): any => {
+				let promisifiedResultFunction: any = decoratorsLib.exportedPromise("moduleName", postAction);
+				// Call this line in order to generate publicApi and get the real Promise
+				promisifiedResultFunction({}, "propertyName", {});
+				return $injector.publicApi.__modules__["moduleName"]["propertyName"]();
+			};
+
+			let assertResults = (result: any): void => {
+				assert.deepEqual(result, expectedResults);
+				assert.isTrue(isPostActionExecuted, "Post action must be executed after all promises are resolved.");
+				assert.isTrue(isActionExecuted, "All actions must be executed after the promise is resolved.");
+			};
+
+			beforeEach(() => {
+				isPostActionExecuted = false;
+				isActionExecuted = false;
+			});
+
+			it("executes postAction after all promises are resolved (function returning IFuture<T>)", () => {
+				expectedResults = "result";
+
+				$injector.register("moduleName", {
+					propertyName: () => {
+						if (isPostActionExecuted) {
+							throw new Error("Post action MUST NOT be called before all actions are executed.");
+						}
+
+						isActionExecuted = true;
+						return Future.fromResult(expectedResults);
+					}
+				});
+
+				let promise: any = getPromisesWithPostAction();
+				promise.then(assertResults);
+			});
+
+			it("executes postAction after all promises are resolved (function returning IFuture<T>[])", () => {
+				let calledActionsCount = 0;
+
+				$injector.register("moduleName", {
+					propertyName: () => _.map(expectedResults, expectedResult => {
+						if (isPostActionExecuted) {
+							throw new Error("Post action MUST NOT be called before all actions are executed.");
+						}
+
+						calledActionsCount++;
+						isActionExecuted = calledActionsCount === expectedResults.length;
+						return Future.fromResult(expectedResult);
+					})
+				});
+
+				expectedResults = ["result1", "result2", "result3"];
```

Comment:
Can you move this line above the `$injector.register` so the logic becomes clearer